float32[3] angles  # rad  target roll angle, pitch angle, and yaw PI term (not target yaw angle)
float32[3] body_rates  # rad/s  target roll rate, pitch rate, yaw rate
float32[] base_thrust  # N
